state-sync/aptos-data-client/src/utils.rs

```diff
@@ -66,33 +70,40 @@ pub fn is_priority_peer(
     false
 }
 
-/// Chooses the peer with the lowest distance from the validator set and
-/// measured latency (using the given set of serviceable peers). We prioritize
-/// distance over latency, since we want to avoid close but not up-to-date peers.
-pub fn choose_lowest_distance_and_latency_peer(
+/// Chooses a peer with the lowest distance from the validator set weighted by
+/// latency (from the given set of peers). We prioritize distance over latency
+/// as we want to avoid close but not up-to-date peers.
+///
+/// Peer selection is done by: (i) identifying all peers with the same lowest
```

Comment:
I wonder if (i) could be too restrictive at some point. I.e., if there's only a single node with lowest distance, and a bunch with lowest distance + 1.

Reply:
Yeah, potentially, but the only real way I could see this happening today is if someone adds a lot of seeds, which should be uncommon 🤞 But, it's good to think about going forward.